tests/test_lint_maturity.py

```diff
@@ -0,0 +1,300 @@
+    "b",
+    "rc",
+    "invalid",
+]
+
+EXPECTED_RELEASE_VERSION = [
+    "stable",
+    "a",
+    "b",
+    "rc",
+    "exception",
+]
+
+INVALID_TAGS = {
+    "a": ["invalid"],
+    "b": ["a", "invalid"],
+    "rc": ["a", "b", "invalid"],
+    "stable": ["a", "b", "rc", "invalid"],
+    "exception": ["a", "b", "rc", "invalid"],
+}
+
+EXPECTED_N_INVALID_TAGS = {
+    "a": 1,
+    "b": 2,
+    "rc": 3,
+    "stable": 4,
+    "exception": 4,
+}
+
+EXPECTED_SYSTEMEXIT = """2020.02.01-py27.yml has 4 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag b packages: [('package_b1', 'v3.1.b1')]
+\tTag rc packages: [('package_rc1', 'v3.1.rc1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+2020.02.a1-py27.yml has 1 packages with invalid maturity tag.
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+2020.02.b1-py27.yml has 2 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+2020.02.rc1-py27.yml has 3 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag b packages: [('package_b1', 'v3.1.b1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+"""
+
+EXPECTED_WARNING = """2020.02.01-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+2020.02.a1-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+2020.02.b1-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+2020.02.rc1-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+bleeding-py27.yml not lint because it is in the exception list.
+bleeding-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+bleeding-py27.yml has 4 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag b packages: [('package_b1', 'v3.1.b1')]
+\tTag rc packages: [('package_rc1', 'v3.1.rc1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+"""
+
+
+def _create_tmp_test_files(release_sample, file_names_sample):
+
+    folder_name = os.getcwd() + "/test_lint_maturity/"
+    os.mkdir(folder_name)
+    list_files = []
+
+    for file_name in file_names_sample:
+        list_files.append(folder_name + file_name)
+
+        with open(folder_name + file_name, "w") as file_sample:
+            file_sample.write(release_sample)
+
+    return list_files
+
+
+@pytest.mark.parametrize(
+    "release_basename, release_version",
+    zip(RELEASE_FILE_NAMES[:-1], EXPECTED_RELEASE_VERSION[:-1]),
+)
+def test_msg_packages_invalid(release_basename, release_version):
+
+    count_tag_invalid = EXPECTED_N_INVALID_TAGS[release_version]
+
+    exit_msg = lint_maturity.msg_packages_invalid(
+        release_basename,
+        release_version,
+        count_tag_invalid,
+        EXPECTED_DICT_TAG_MATURITY,
+    )
+
+    assert exit_msg in EXPECTED_SYSTEMEXIT
```

Comment:
This is a general comment that goes for all your tests, I think they would be much easier to read if you change them to be more explicit, like this:
```suggestion
@pytest.mark.parametrize(
    "release_basename, release_version, count_tag_invalid",
    [
        ('2020.02.01-py27.yml', 'stable', 4),
        ('2020.02.a1-py27.yml', 'a', 1),
        ('2020.02.b1-py27.yml', 'b', 2),
        ('2020.02.rc1-py27.yml', 'rc', 3)
    ]
)
def test_msg_packages_invalid(release_basename, release_version, count_tag_invalid):

    exit_msg = lint_maturity.msg_packages_invalid(
        release_basename,
        release_version,
        count_tag_invalid,
        EXPECTED_DICT_TAG_MATURITY,
    )

    assert exit_msg in EXPECTED_SYSTEMEXIT
``
I would even prefer including `EXPECTED_DICT_TAG_MATURITY` and `EXPECTED_SYSTEMEXIT`, but that might be harder. It makes it much easier to read what you are actually testing.

Reply:
Thanks! 
Even if the same structure/data is used multiple times in the code?
I thought it would be too much wordy to be explicit for some values, e.g. RELEASE_FILE_NAMES is used 7 times, EXPECTED_RELEASE_VERSION is used 8, TAG_EXCEPTIONS 16 times.
Would it be ok if I only let explicit what is used only once?

Reply:
In terms of `EXPECTED_DICT_TAG_MATURITY` and `EXPECTED_SYSTEMEXIT` it makes more sense to have global ones. Is it possible to make the tests even more generic though, so they dont expect the same? That might not make sense though so feel free to disregard.

Reply:
I was a little skeptical when I read but I now think your suggestion was very good. many thanks!
After the review, what is tested is closer to the tests and I also add the name of the arguments explicitly in some functions to make it easier to read.
It is perhaps a little wordy, but that is because we have a test per type of release (stable, a,b,rc, ...) most of the time.